packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraUtils.java

```diff
@@ -25,61 +24,14 @@
 
   private CameraUtils() {}
 
-  static Size[] computeBestPreviewAndRecordingSize(
-      Activity activity,
-      StreamConfigurationMap streamConfigurationMap,
-      int minHeight,
-      int orientation,
-      Size captureSize) {
-    Size previewSize, videoSize;
-    Size[] sizes = streamConfigurationMap.getOutputSizes(SurfaceTexture.class);
-
-    // Preview size and video size should not be greater than screen resolution or 1080.
-    Point screenResolution = new Point();
-
-    Display display = activity.getWindowManager().getDefaultDisplay();
-    display.getRealSize(screenResolution);
-
-    final boolean swapWH = orientation % 180 == 90;
-    int screenWidth = swapWH ? screenResolution.y : screenResolution.x;
-    int screenHeight = swapWH ? screenResolution.x : screenResolution.y;
-
-    List<Size> goodEnough = new ArrayList<>();
-    for (Size s : sizes) {
-      if (minHeight <= s.getHeight()
-          && s.getWidth() <= screenWidth
-          && s.getHeight() <= screenHeight
-          && s.getHeight() <= 1080) {
-        goodEnough.add(s);
-      }
+  static Size computeBestPreviewSize(String cameraName, ResolutionPreset preset) {
+    if (preset.ordinal() > ResolutionPreset.high.ordinal()) {
+      preset = ResolutionPreset.high;
```

Comment:
Why do you set the preset to high here?

Reply:
i think it's that high res previews negatively impact performance. The current code in master is setting previewSize to be the absolute minimum resolution that's the correct aspect ratio, and is also capped at 1080 (line 37 in the original patch).

Reply:
Maybe you could improve to let the choice to the user here. My current use case need a stream at the highest resolution. And without this clamp, the function startPreviewWithImageStream works perfectly.

Reply:
It would be acceptable to set the default to `high` or even `medium`. But this change actually cripples the powerful Android devices that default to 1080p video even at 60fps.